edenai_apis/apis/sightengine/sightengine_api.py

```diff
@@ -0,0 +1,171 @@
+import json
+from typing import Dict, Any, Optional
+import requests
+from edenai_apis.apis.amazon.helpers import check_webhook_result
+from edenai_apis.features import ProviderInterface, ImageInterface
+
+from edenai_apis.features.video.deepfake_detection_async.deepfake_detection_async_dataclass import (
+    DeepfakeDetectionAsyncDataClass as VideoDeepfakeDetectionAsyncDataclass,
+)
+from edenai_apis.features.image.deepfake_detection.deepfake_detection_dataclass import (
+    DeepfakeDetectionDataClass as ImageDeepfakeDetectionDataclass,
+)
+from edenai_apis.loaders.data_loader import ProviderDataEnum
+from edenai_apis.loaders.loaders import load_provider
+from edenai_apis.utils.exception import ProviderException
+from edenai_apis.utils.parsing import extract
+from edenai_apis.utils.types import AsyncBaseResponseType, AsyncLaunchJobResponseType, AsyncPendingResponseType, AsyncResponseType, ResponseType
+from edenai_apis.utils.upload_s3 import upload_file_to_s3
+
+class SightEngineApi(ProviderInterface, ImageInterface):
+    provider_name = "sightengine"
+
+    def __init__(self, api_keys: Optional[Dict[str, Any]] = None):
+        self.api_settings = load_provider(
+            ProviderDataEnum.KEY,
+            provider_name=self.provider_name,
+            api_keys=api_keys or {},
+        )
+        self.api_url = "https://api.sightengine.com/1.0"
+        self.headers = {
+            "Content-Type": "application/json",
+            "Authorization": f'Bearer {self.api_settings["api_key"]}',
+            
+        }
+        self.webhook_settings = load_provider(ProviderDataEnum.KEY, "webhooksite")
+        self.webhook_token = self.webhook_settings["webhook_token"]
+        self.webhook_url = f"https://webhook.site/{self.webhook_token}"
+
+    def image__deepfake_detection(
+        self, file: Optional[str] = None, file_url: Optional[str] = None
+    ) -> ResponseType[ImageDeepfakeDetectionDataclass]:
+        if not file_url and not file:
+            raise ProviderException("file or file_url required")
+
+        payload = {"url": file_url or upload_file_to_s3(file, file),
+                              "models":"deepfake",
+                              "api_user": self.api_settings["api_user"],
+                              "api_secret": self.api_settings["api_key"]}
+        
+        response = requests.get(

```

Comment:
usually I like to try surround the requests call with a try/catch bloc in case of some exceptions like Timeout or ConnectionError, and than in case of an exception, raise a ProviderException. Same for the deepfake video